### PR TITLE
bin: log panic info if possible

### PR DESCRIPTION
**Pull request by BusyJay**
ref #1129 

@ngaut @siddontang @disksing @zhangjinpeng1987 PTAL

### CODE REVIEW

File: src/util/panic_hook.rs [hunk 1]
```diff
@@ -65,7 +68,28 @@ fn track_hook(p: &PanicInfo) {
 pub fn set_exit_hook() {
     let orig_hook = panic::take_hook();
     panic::set_hook(box move |info: &PanicInfo| {
```
Comment (ngaut) [inline src/util/panic_hook.rs]: What's the behavior if the log level is info or warn ?

Comment (BusyJay) [inline src/util/panic_hook.rs]: It will log the panic message and backtrace to the file at ERROR level.

Comment (ngaut) [inline src/util/panic_hook.rs]: Great.

File: src/util/panic_hook.rs [hunk 2]
```diff
@@ -65,7 +68,28 @@ fn track_hook(p: &PanicInfo) {
...
-        orig_hook(info);
+        if log_enabled!(LogLevel::Error) {
+            let msg = match info.payload().downcast_ref::<&'static str>() {
```
Comment (siddontang) [inline src/util/panic_hook.rs]: can we extract these to a function and test?

Comment (BusyJay) [inline src/util/panic_hook.rs]: The log output file is global and can't be changed outside log crate after being initialized.

**Comment by siddontang** [comment]
LGTM

**Comment by ngaut** [comment]
LGTM

**Pull request merged**